### ISSUE TITLE
H2 db lost data during traccar service stop/restart

### ISSUE THREAD

**Issue by ulion** [traccar/traccar#1479]
```
2015-10-28 10:42:49  INFO: Shutting down server...
2015-10-28 10:42:52  INFO: Operating system name: Mac OS X version: 10.10.5 architecture: x86_64
2015-10-28 10:42:52  INFO: Java runtime name: Java HotSpot(TM) 64-Bit Server VM vendor: Oracle Corporation version: 25.45-b02
2015-10-28 10:42:52  INFO: Memory limit heap: 1820mb non-heap: 0mb
2015-10-28 10:42:52  INFO: Character encoding: UTF-8 charset: UTF-8
2015-10-28 10:42:52  INFO: Version: null
2015-10-28 10:42:54  INFO: Starting server...
2015-10-28 10:43:06  INFO: Shutting down server...
2015-10-28 10:43:08  INFO: Operating system name: Mac OS X version: 10.10.5 architecture: x86_64
2015-10-28 10:43:08  INFO: Java runtime name: Java HotSpot(TM) 64-Bit Server VM vendor: Oracle Corporation version: 25.45-b02
2015-10-28 10:43:08  INFO: Memory limit heap: 1820mb non-heap: 0mb
2015-10-28 10:43:08  INFO: Character encoding: UTF-8 charset: UTF-8
2015-10-28 10:43:08  INFO: Version: null
2015-10-28 10:43:10  INFO: Starting server...
```
I installed the traccar as service, but once that is done and system reboot, I can not stop the traccar by the traccar stop command, the wrapper restarted the traccar at once it is stopped by my command.
anyway, I called traccar stop 3 times, then finally when it auto restart, I found the h2 database lost recent 5 days data. lost db data is a really bad thing!!

what I done in terminal:

```
localhost:huadefu ulion$ ps -ef|grep trac
    0    96     1   0 五09上午 ??        16:43.40 /bin/sh /opt/traccar/bin/traccar launchdinternal
    0   380     1   0 五09上午 ??         9:29.01 /opt/traccar/bin/./wrapper /opt/traccar/bin/../conf/wrapper.conf wrapper.syslog.ident=traccar wrapper.pidfile=/opt/traccar/bin/./traccar.pid wrapper.name=traccar wrapper.displayname=traccar wrapper.daemonize=TRUE wrapper.statusfile=/opt/traccar/bin/./traccar.status wrapper.java.statusfile=/opt/traccar/bin/./traccar.java.status wrapper.script.version=3.5.26
    0   393   380   0 五09上午 ??        38:12.22 /usr/bin/java -d64 -Djava.library.path=../lib -classpath ../lib/wrapper.jar:../tracker-server.jar -Dwrapper.key=xXu3RwXZtzvfKnd3 -Dwrapper.port=32000 -Dwrapper.disable_console_input=TRUE -Dwrapper.pid=380 -Dwrapper.version=3.5.26 -Dwrapper.native_library=wrapper -Dwrapper.arch=universal -Dwrapper.service=TRUE -Dwrapper.cpu.timeout=10 -Dwrapper.jvmid=1 org.tanukisoftware.wrapper.WrapperSimpleApp org.traccar.Main ../conf/traccar.xml
  501 31264   862   0 10:42上午 ttys000    0:00.00 grep trac
  501 59494   894   0 二08上午 ttys002    0:00.14 vi /opt/traccar/logs/tracker-server.log
localhost:huadefu ulion$ sudo /opt/traccar/bin/traccar stop
Password:
Stopping traccar...
localhost:huadefu ulion$ ps -ef|grep trac
    0    96     1   0 五09上午 ??        16:43.44 /bin/sh /opt/traccar/bin/traccar launchdinternal
    0   380     1   0 五09上午 ??         9:29.03 /opt/traccar/bin/./wrapper /opt/traccar/bin/../conf/wrapper.conf wrapper.syslog.ident=traccar wrapper.pidfile=/opt/traccar/bin/./traccar.pid wrapper.name=traccar wrapper.displayname=traccar wrapper.daemonize=TRUE wrapper.statusfile=/opt/traccar/bin/./traccar.status wrapper.java.statusfile=/opt/traccar/bin/./traccar.java.status wrapper.script.version=3.5.26
    0   393   380   0 五09上午 ??        38:12.52 /usr/bin/java -d64 -Djava.library.path=../lib -classpath ../lib/wrapper.jar:../tracker-server.jar -Dwrapper.key=xXu3RwXZtzvfKnd3 -Dwrapper.port=32000 -Dwrapper.disable_console_input=TRUE -Dwrapper.pid=380 -Dwrapper.version=3.5.26 -Dwrapper.native_library=wrapper -Dwrapper.arch=universal -Dwrapper.service=TRUE -Dwrapper.cpu.timeout=10 -Dwrapper.jvmid=1 org.tanukisoftware.wrapper.WrapperSimpleApp org.traccar.Main ../conf/traccar.xml
  501 31433   862   0 10:42上午 ttys000    0:00.00 grep trac
localhost:huadefu ulion$ sudo /opt/traccar/bin/traccar stop
Stopping traccar...
localhost:huadefu ulion$ sudo /opt/traccar/bin/traccar stop
Stopping traccar...
traccar is not running.
localhost:huadefu ulion$ ps -ef|grep trac
    0 31758     1   0 10:43上午 ??         0:00.03 /bin/sh /opt/traccar/bin/traccar launchdinternal
    0 31879     1   0 10:43上午 ??         0:00.01 /opt/traccar/bin/./wrapper /opt/traccar/bin/../conf/wrapper.conf wrapper.syslog.ident=traccar wrapper.pidfile=/opt/traccar/bin/./traccar.pid wrapper.name=traccar wrapper.displayname=traccar wrapper.daemonize=TRUE wrapper.statusfile=/opt/traccar/bin/./traccar.status wrapper.java.statusfile=/opt/traccar/bin/./traccar.java.status wrapper.script.version=3.5.26
    0 31880 31879   0 10:43上午 ??         0:03.18 /usr/bin/java -d64 -Djava.library.path=../lib -classpath ../lib/wrapper.jar:../tracker-server.jar -Dwrapper.key=XgcxXoRyTkM9TihD -Dwrapper.port=32000 -Dwrapper.disable_console_input=TRUE -Dwrapper.pid=31879 -Dwrapper.version=3.5.26 -Dwrapper.native_library=wrapper -Dwrapper.arch=universal -Dwrapper.service=TRUE -Dwrapper.cpu.timeout=10 -Dwrapper.jvmid=1 org.tanukisoftware.wrapper.WrapperSimpleApp org.traccar.Main ../conf/traccar.xml
  501 31894   862   0 10:43上午 ttys000    0:00.00 grep trac
```

**Comment by tananaev** [comment]
I suspect there is some problem with Java Service Wrapper. I don't think Mac is their main target platform, so there might be some issues especially on new versions of OS X.

As for lost data, are you sure it was there before and disappeared after restart? If that's the case I guess DB file got corrupted. I would recommend to use regular database (e.g. MySQL) for any purposes other than trying out.

**Comment by ulion** [comment]
Thank you~

So there are 2 problems:

1. after the call traccar install and restart system, the traccar service
can not be stopped by traccar stop command, the wrapper always try to
restart the service.
2. with above case, stop the service may cause h2 database lost memory data.

2015-10-28 13:00 GMT+08:00 Anton Tananaev <notifications@github.com>:

> I suspect there is some problem with Java Service Wrapper. I don't think
> Mac is their main target platform, so there might be some issues especially
> on new versions of OS X.
>
> As for lost data, are you sure it was there before and disappeared after
> restart? If that's the case I guess DB file got corrupted. I would
> recommend to use regular database (e.g. MySQL) for any purposes other than
> trying out.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tananaev/traccar/issues/1479#issuecomment-151726024>.
>



-- 
Ulion

**Comment by tananaev** [comment]
1. Hopefully this problem will be fixed with the next version of Java Service Wrapper. I couldn't find the issue in their bug tracker, but you can try to report your problem there - https://sourceforge.net/p/wrapper/bugs/. I can assist with some technical questions if they have any.

2. I guess the corrupted database issue is cause by the shutdown problem.

**Comment by ulion** [comment]
In the newest wrapper shell script, I found following code part:
```
+            eval echo `gettext 'Daemon is set to be kept continuously running and it will be automatically restarted.'`
+            eval echo `gettext 'To stop the daemon you need to uninstall it.'`
+            eval echo `gettext 'If you want to use the \"stop\" argument, you need to find MACOSX_KEEP_RUNNING'`
+            eval echo `gettext 'at the beginning of the script file and set it to \"false\".'`
```

So, the traccar script need be upgraded to newer version from the wrapper source repo, maybe also other related files
ref: http://sourceforge.net/p/wrapper/code/HEAD/tree/trunk/wrapper/src/bin/sh.script.in

**Comment by tananaev** [comment]
I looked through the changes and it seems like the OS X problem is fixed in version 3.5.27 of Java Service Wrapper, so next release of Traccar server should work fine.
